src/shared/generator.c

```diff
@@ -188,7 +188,7 @@ int generator_write_timeouts(
 
         return write_drop_in_format(dir, unit, 50, "device-timeout",
                                     "# Automatically generated by %s\n\n"
-                                    "[Unit]\nJobTimeoutSec=%s",
+                                    "[Unit]\nJobRunningTimeoutSec=%s",
                                     program_invocation_short_name, timeout);
```

Comment:
The two (finite) timeouts will always have a common end. JobTimeoutSec= is
spanning the JobRunningTimeoutSec= so there is only one timer event created for
the timeout that goes off earlier. (I find two events couter-productive if
equal/close values were configured for both.)